docs/flag_source_configuration.md

```diff
@@ -54,6 +56,8 @@ The relevant `FlagSourceConfigurations` are passed to the operator by setting th
 | DefaultSyncProvider | Defines the default provider to be used, can be set to `kubernetes`, `filepath`, `http` or `flagd-proxy` (experimental).                                                                                                  | optional `string`                                                         | `kubernetes`                 | 
 | RolloutOnChange     | When set to true the operator will trigger a restart of any `Deployments` within the `FlagSourceConfiguration` reconcile loop, updating the injected sidecar with the latest configuration. | optional `boolean`                                                        | `false`                      | 
 | ProbesEnabled       | Enable or disable Liveness and Readiness probes of the flagd sidecar. When enabled, HTTP probes( paths - `/readyz`, `/healthz`) are set with an initial delay of 5 seconds                  | optional `boolean`                                                        | `true`                       |       
+| DebugLogging        | Enable or disable --debug flag of flagd sidecar                                                                                                                                             | optional `boolean`                                                        | `false`                      |
+| RawSidecarArgs      | String arguments passed to the flagd sidecar on startup, flagd documentation can be found [here](https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md)       | optional `array of strings`                                               | `""`                         |
```

Comment:
I tried applying multiple parameters to the flagD binary and the current behaviour is that the latest applied value wins

```
```

Reply:
Let's add them as the first thing and document that any direct value configured wins over the value set via the `RawSidecarArgs` flag

Reply:
Actually we probably need to have a controlling logic in the operator. Let's take a use-case, when the user sets the `.spec.debugLogging` field to false (or does not set it at all and the default value is false) and afterwards the `.spec.rawSidecarArgs` contain `--debug` flag, which is basically the flag set when `.spec.debugLogging` is set to true. In this case, we need to parse the `.spec.rawSidecarArgs` and check for inconsistent states. 

In the current state I am actually questioning my original idea of adding `.spec.rawSidecarArgs` to the FlagSourceConfiguration CRD and if it's worth the work. What about your thoughts @toddbaert @thisthat ?

Reply:
IMO, adding `DebugLogging` is essential. However, given the complexity of parsing & validations, may be it is better to avoid `rawSidecarArgs ` 🤔  If we really want to control the flagd startup flags, then it is best exposed through CRD, so that we know how to control that flag.

Reply:
already reverted the changes of `rawSidecarArgs` (last commit in this PR), but still waiting for input from @toddbaert

Reply:
Ya, this is fine with me. It was a simple nice-to-have, but seems like it's added too much complexity.